pom.xml

```diff
@@ -798,6 +784,33 @@
               </execution>
             </executions>
           </plugin>
+          <plugin>
+            <!-- hacky META-INF/versions shade relocation -->
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <version>3.1.0</version>
+            <executions>
+              <execution>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <phase>package</phase>
+                <configuration>
+                  <target>
+                    <unzip dest="${project.build.directory}/tmp" src="${project.build.directory}/${project.build.finalName}.jar"/>
+                    <mkdir dir="${project.build.directory}/tmp/META-INF/versions/9/net/snowflake/ingest/shade/net/snowflake/client"/>
+                    <mkdir dir="${project.build.directory}/tmp/META-INF/versions/11/net/snowflake/ingest/shade/net/snowflake/client"/>
+                    <mkdir dir="${project.build.directory}/tmp/META-INF/versions/15/net/snowflake/ingest/shade/net/snowflake/client"/>
+                    <move file="${project.build.directory}/tmp/META-INF/versions/9/net/snowflake/client" todir="${project.build.directory}/tmp/META-INF/versions/9/net/snowflake/ingest/shade/net/snowflake/"/>
+                    <move file="${project.build.directory}/tmp/META-INF/versions/11/net/snowflake/client" todir="${project.build.directory}/tmp/META-INF/versions/9/net/snowflake/ingest/shade/net/snowflake/"/>
+                    <move file="${project.build.directory}/tmp/META-INF/versions/15/net/snowflake/client" todir="${project.build.directory}/tmp/META-INF/versions/9/net/snowflake/ingest/shade/net/snowflake/"/>
+                    <zip basedir="${project.build.directory}/tmp" destfile="${project.build.directory}/${project.build.finalName}.jar"/>
+                    <delete dir="${project.build.directory}/tmp"/>
+                  </target>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
```

Comment:
The relocation plugin doesn't handle multi-release jars, but https://stackoverflow.com/questions/74119618/maven-shade-plugin-relocate-classes-under-meta-inf-versions-in-multi-release-jar provides a hacky workaround